### PR TITLE
Added Stack Data Structures.md 

### PR DESCRIPTION
**Pull request by jo780-full**
Stacks Data structure is added

### CODE REVIEW

**Comment by jo780-full** [comment]
Please find the issue i raised for this pr

**Comment by jo780-full** [comment]
okay will do it

**Comment by jo780-full** [comment]
kindly do check now i have made changes as required

File: en/Data Structures/Stacks/stack.md [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
 Some Standard Stack Operations are:
-push() : inserts the element at the top of the stack.
+push(): inserts the element at the top of the stack.
 pop(): When we delete the element at the top of the stack
-isEmpty():It determines whether the stack is empty.
-isFull():determines the stack is full or not
+isEmpty(): It determines whether the stack is empty.
+isFull(): determines whether the stack is full or not
+peek(): Gets the value of the top element without removing it
+count(): It returns the total number of elements available in a stack.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:14]: `count`, `change` and `display` are probably out of scope here

File: en/Data Structures/Stacks/stack.md [hunk 2]
```diff
@@ -1,19 +1,34 @@
 # Stack
 
-Stack is a basic linear data structure which follows an order in which operations are perfomed.The order is called LIFO(Last In First Out)or FILO.
-A perfect example for a stack of plates in a canteen,a pile of books,a box of pringles.
-Stacks are used to implement parsers and evalution expressions and backtracking algorithms. basic operations are pushing an element into stack and popping the element out of the stack.
+A stack is a basic linear data structure that follows an order in which operations are performed. The order is called LIFO(Last In First Out)or FILO.
+A perfect example of stacks would be plated in a canteen, a pile of books, or a box of Pringles.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:4]: plates*

File: en/Data Structures/Stacks/stack.md [hunk 3]
```diff
@@ -1,19 +1,34 @@
 # Stack
 
-Stack is a basic linear data structure which follows an order in which operations are perfomed.The order is called LIFO(Last In First Out)or FILO.
-A perfect example for a stack of plates in a canteen,a pile of books,a box of pringles.
-Stacks are used to implement parsers and evalution expressions and backtracking algorithms. basic operations are pushing an element into stack and popping the element out of the stack.
+A stack is a basic linear data structure that follows an order in which operations are performed. The order is called LIFO(Last In First Out)or FILO.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:3]: I wouldn't call this the "order in which operations are performed"; rather it's the order in which objects can be accessed:

LIFO: The last object to be pushed to the stack is the first object to be popped;
FILO: The first object to be pushed to the stack is the last object to be popped

File: en/Data Structures/Stacks/stack.md [hunk 4]
```diff
@@ -1,19 +1,34 @@
...
+A perfect example of stacks would be plated in a canteen, a pile of books, or a box of Pringles.
+Stacks are used to implement parsers and evaluation expressions and backtracking algorithms. basic operations are pushing an element into the stack and popping the element out of the stack.
 We can make use of linked lists or arrays of lists. The stack contains only one pointer
-"top pointer" which points to the top most elements of the stack.Insertion and deletion occurs at only one end of the stack.
+"top pointer" which points to the topmost elements of the stack. Insertion and deletion occur at only one end of the stack.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:7]: only occur at*

**Comment by jo780-full** [comment]
Done!!

File: en/Data Structures/Stacks/stack.md [hunk 5]
```diff
@@ -11,17 +11,15 @@ pop(): When we delete the element at the top of the stack
...
-    A pointer called TOP is used to keep track of the top element in the stack.
-    When initializing the stack, we set its value to -1 so that we can check if the stack is empty by comparing TOP == -1.
-    On pushing an element, we increase the value of TOP and place the new element in the position pointed to by TOP.
-    On popping an element, we return the element pointed to by TOP and reduce its value.
-    Before pushing, we check if the stack is already full
-    Before popping, we check if the stack is already empty
+ 1.   A pointer called TOP is used to keep track of the top element in the stack.
+ 2.  When initializing the stack, we set its value to -1 so that we can check if the stack is empty by comparing TOP == -1.
+ 3.   On pushing an element, we increase the value of TOP and place the new element in the position pointed to by TOP.
+ 4.   On popping an element, we return the element pointed to by TOP and reduce its value.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:20]: Why 3 spaces (rather than one) here?

Comment (jo780-full) [reply en/Data Structures/Stacks/stack.md]: It is changed. I also changed the video

File: en/Data Structures/Stacks/stack.md [hunk 6]
```diff
@@ -14,20 +14,21 @@ peek(): Gets the value of the top element without removing it
 
 
 Working with Stacks
- 1.   A pointer called TOP is used to keep track of the top element in the stack.
- 2.  When initializing the stack, we set its value to -1 so that we can check if the stack is empty by comparing TOP == -1.
- 3.   On pushing an element, we increase the value of TOP and place the new element in the position pointed to by TOP.
- 4.   On popping an element, we return the element pointed to by TOP and reduce its value.
- 5.   Before pushing, we check if the stack is already full
- 6.   Before popping, we check if the stack is already empty
+ 1.A pointer called TOP is used to keep track of the top element in the stack.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md]: There should be exactly one space after the dot.

File: en/Data Structures/Stacks/stack.md [hunk 7]
```diff
@@ -14,12 +14,12 @@ peek(): Gets the value of the top element without removing it
 
 
 Working with Stacks
- 1)A pointer called TOP is used to keep track of the top element in the stack.
- 2)When initializing the stack, we set its value to -1 so that we can check if the stack is empty by comparing TOP == -1.
- 3)On pushing an element, we increase the value of TOP and place the new element in the position pointed to by TOP.
- 4)On popping an element, we return the element pointed to by TOP and reduce its value.
- 5)Before pushing, we check if the stack is already full
- 6)Before popping, we check if the stack is already empty
+ 1) A pointer called TOP is used to keep track of the top element in the stack.
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:17]: well now it's not a Markdown list anymore, is it? is this intended?

Comment (jo780-full) [reply en/Data Structures/Stacks/stack.md:17]: yea it is intended

Comment (appgurueu) [reply en/Data Structures/Stacks/stack.md:17]: this may cause poor line wrapping though

File: en/Data Structures/Stacks/stack.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# Stack
+
+A stack is a basic linear data structure that follows an order in which objects are accessed. The order is called LIFO(Last In First Out)or FILO(First in Last Out).
+A perfect example of stacks would be plates in a canteen, a pile of books, or a box of Pringles,etc.
+Stacks are used to implement parsers and evaluation expressions and backtracking algorithms. basic operations are pushing an element into the stack and popping the element out of the stack.
+We can make use of linked lists or arrays of lists. The stack contains only one pointer
+"top pointer" which points to the topmost elements of the stack. Insertion and deletion  only occurs at one end of the stack.
+# Standard Stack Operations are:
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:8]: Just `# Standard Stack Operations` (also: please add blank lines around headings).

File: en/Data Structures/Stacks/stack.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+2) pop(): When we delete the element at the top of the stack
+3) isEmpty(): It determines whether the stack is empty.
+4) isFull(): determines whether the stack is full or not
+5) peek(): Gets the value of the top element without removing it
+
+
+# Working with Stacks
+A pointer called TOP is used to keep track of the top element in the stack.When initializing the stack, we set its value to -1 so that we can check if the stack is empty by comparing TOP == -1.On pushing an element, we increase the value of TOP and place the new element in the position pointed to by TOP.On popping an element, we return the element pointed to by TOP and reduce its value.Before pushing, we check if the stack is already fullBefore popping, we check if the stack is already empty
+
+
```
Comment (appgurueu) [inline en/Data Structures/Stacks/stack.md:19]: Why three blank lines?

**Review by appgurueu — COMMENTED** [review]
Some remarks regarding punctuation:
- Please always leave a space before `(`
- Please always leave a space after `.`, `,` etc.

**Review by appgurueu — DISMISSED** [review]

**Review by appgurueu — DISMISSED** [review]

File: en/Data Structures/Stacks/stack.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+# Source
+
+1) [Stack Data Structure - GeeksForGeeks](https://www.geeksforgeeks.org/stack-data-structure/) 
+2) [DS-Stack JavaPoint](https://www.javatpoint.com/data-structure-stack)
+3) [Stack Data Structure](https://www.programiz.com/dsa/stack)
+
+# Video Playlist
+
+-  [Stack Data Structure](https://youtu.be/F1F2imiOJfk)
```
Comment (Panquesito7) [inline en/Data Structures/Stacks/stack.md:29]:
```suggestion
- [Stack Data Structure](https://youtu.be/F1F2imiOJfk)
```

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by appgurueu — APPROVED** [review]

**Comment by jo780-full** [comment]
Thank you

**Pull request merged**